src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AspNetCore6Plus/BrowserInjectingStreamWrapper.cs

```diff
@@ -65,36 +66,43 @@ public override void SetLength(long value)
 
         public override void Write(byte[] buffer, int offset, int count)
         {
-            if (IsHtmlResponse())
-            {
-                var curBuf = buffer.AsMemory(offset, count).ToArray();
-                _agent.TryInjectBrowserScriptAsync(_context.Response.ContentType, _context.Request.Path.Value, curBuf,
-                        _baseStream)
-                    .GetAwaiter().GetResult();
-            }
-            else
+            if (_context.Items[InjectingRUM] == null) // pass through without modification if we're already in the middle of injecting
```

Comment:
Is it possible to get here without this key being set?

Reply:
This implementation is probably going to be worthy of some explanation at some point - there are some odd "recursion" cases that we have to deal with, necessitating the `_context.Items[]` usage.

Reply:
I think it's fine to use `_context.Items`, but I'm more concerned about causing a crash if the key doesn't exist in that collection. In other words, can we safely use `[InjectingRUM]` to do the check, or do we need to use a different method that will not throw an exception?

Reply:
`.ContainsKey()` I guess is an alternative. Though the current code doesn't throw if the key doesn't exist (it, probably, creates a key with a `null` default value).

Reply:
Reworked the code a bit.